Plugins/Damage/Damage.cpp

```diff
@@ -238,6 +238,7 @@ ArgumentStack Damage::DealDamage(ArgumentStack&& args)
         positive[k] = vDamage[k] > 0;
     }
     int damagePower = Services::Events::ExtractArgument<int32_t>(args);
+    int range = Services::Events::ExtractArgument<int>(args);
```

Comment:
Maybe this should be in a try/catch, so that if something is not pushed (because of old nss file) it doesn't break existing code, but just print a warning and assume 0?

Reply:
```suggestion
    int range = 0;
    try {
        range = Services::Events::ExtractArgument<int>(args);
    }
    catch(std::exception e) {
        LOG_WARNING("NWNX_Damage_DealDamage() called from NWScript without final parameter. Please download the latest versions of NWNX scripts.");
    }
```

Is there any sense to outputting the actual exception text, since there's more specific information available and the problem is readily apparent?

Reply:
Should be fine without the actual exception text